### PR TITLE
Fix #400: move autoloader to src/Google

### LINKED ISSUE #400: Move autoloader to Google/ directory
In Fedora, we have the library packaged below `/usr/share/php/Google` . With the old arrangement where the library itself and things that used it would do something like this:

    require_once Google/Utils.php;

that worked fine - as /usr/share/php is on the system include path, anything that used the library would find the system-wide copy if it was installed. However, the new autoload mechanism is hardcoded to expect the autoloader to be two levels *beneath* the Google directory, stuff like:

    require_once realpath(dirname(__FILE__) . '/../../../autoload.php');

That means we'd have to ship it in `/usr/share`, which is obviously not a good design.

Is there any reason the autoloader can't just be in the Google directory? It'd make things easier.

### PR DESCRIPTION
**Pull request by AdamWill**
This makes system-wide installation of the library more viable.

It's a quick job, but it looks fine. I ran the test suite and got the same results before and after.

### CODE REVIEW

**Comment by AdamWill** [comment]
"On the off chance I had a check, but didn't get a hit on owncloud on the corp signers list. "

I don't work for ownCloud, I work for Red Hat (I package ownCloud for Fedora as a side project). I'm trying to find out if RH legal has an opinion on the individual CLA and/or we have any corporate agreement ATM.

**Comment by ianbarber** [comment]
Sounds good!

File: autoload.php [hunk 1]
```diff
@@ -15,19 +15,12 @@
...
+// PHP 5.2 compatibility: E_USER_DEPRECATED was added in 5.3
+if (!defined('E_USER_DEPRECATED')) {
+  define('E_USER_DEPRECATED', E_USER_WARNING);
 }
 
-spl_autoload_register('google_api_php_client_autoload');
+$error = "google-api-php-client's autoloader was moved to src/Google/autoload.php in 1.1.3. This ";
+$error .= "redirect will be removed in 1.2. Please adjust your code to use the new location.";
+trigger_error($error, E_USER_DEPRECATED);
+require_once 'src/Google/autoload.php';
```
Comment (glensc) [inline autoload.php]: i suggest not  to rely on relative paths, as those get resolved from include_path. 

use sth like this instead:
```php
require_once  dirname(__FILE__) . '/src/Google/autoload.php';
```

**Comment by glensc** [comment]
how about dropping "require_once" lines from classes? the documentation already says you should first load autoloader, meaning you do not need to do that again from classes themselves.

**Comment by AdamWill** [comment]
@glensc AIUI, the `require_once` are there for backwards compatibility, to make existing projects (which directly require the files they need, as was previously the only option) keep working with the new version.

File: autoload.php (hunk 1 above)
Comment (AdamWill) [inline autoload.php]: good point, thanks.

**Comment by ianbarber** [comment]
Any joy on the CLA?

**Comment by AdamWill** [comment]
Sorry, still didn't hear back from the right person - I'm gonna give it a day or two then just go ahead and sign the personla version.

**Comment by AdamWill** [comment]
I signed the individual CLA today, sorry for the delay!

**Comment by ianbarber** [comment]
Thanks!

**Pull request merged**